subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformParallelIntegrationTest.groovy

```diff
@@ -102,6 +121,50 @@ class ArtifactTransformParallelIntegrationTest extends AbstractDependencyResolut
         outputContains("Transforming test3-3.3.jar to test3-3.3.jar.txt")
     }
 
+    def "transformations are applied in parallel for project artifacts"() {
+        given:
+        settingsFile << """
+            include "lib1", "lib2", "lib3"
+        """
+
+        buildFile << """            
+            configure([project(":lib1"), project(":lib2"), project(":lib3")]) {
+
+                task jar(type: Jar) {
+                    archiveName = "\${project.name}.jar"
+                    destinationDir = buildDir
+                }
+                artifacts {
+                    compile jar
+                }
+            }
+
+            dependencies {
+                compile project(":lib1")
+                compile project(":lib2")
+                compile project(":lib3")
+            }
+            task resolve {
+                doLast {
+                    def artifacts = configurations.compile.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    println artifacts.artifactFiles.collect { it.name }
```

Comment:
Is this `println` required? If it's only required so we can see `Transforming ...` as well, then maybe we could print something here and look for that something in the output instead?

Reply:
I changed the `println` to an `assert`.